### PR TITLE
Add company controller

### PR DESCRIPTION
**Pull request by Eason-Gao**
Added delete controller as part of the ticket

### CODE REVIEW

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -1,2 +1,38 @@
 class CompaniesController < ApplicationController
+  def index
+    @companies = Company.all
+    render html: "TODO: company display page"
```
Comment (ryan-mok) [inline app/controllers/companies_controller.rb]: Perhaps considering doing the index page also as part of this PR?

File: db/schema.rb [hunk 2]
```diff
@@ -1,50 +1,50 @@
...
+  create_table "users", force: :cascade do |t|
+    t.string "first_name"
+    t.string "last_name"
+    t.string "email"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.string "password_digest"
+    t.index ["email"], name: "index_users_on_email", unique: true
+  end
+
```
Comment (zhangium) [inline db/schema.rb]: Stray newline

File: app/controllers/companies_controller.rb [hunk 3]
```diff
@@ -1,2 +1,38 @@
...
+    if @company.update_attributes(company_params)
+      redirect_to @company
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
+  def company_params
```
Comment (zhangium) [inline app/controllers/companies_controller.rb]: indent private methods

File: app/controllers/companies_controller.rb (hunk 1 above)
Comment (zhangium) [reply app/controllers/companies_controller.rb]: agree, should just do it in this PR if possible

Comment (ryan-mok) [reply app/controllers/companies_controller.rb]: I have an idea on how to implement the index page after working on the autocompletion. If this work hasn't been done yet, I can do this work on a separate branch after this is merged.

File: app/controllers/companies_controller.rb (hunk 3 above)
Comment (Eason-Gao) [reply app/controllers/companies_controller.rb]: done

File: app/assets/stylesheets/companies.scss [hunk 4]
```diff
@@ -1,3 +1,11 @@
...
+
+.companies {
+  list-style: none;
+  margin: 0;
+  li {
+    overflow: auto;
+    padding: 10px 0;
+    border-bottom: 1px solid lightgrey;
+  }
+}
```
Comment (ryan-mok) [inline app/assets/stylesheets/companies.scss]: Newline character at end of files

File: app/views/companies/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+<h1>All Companies </h1>
+
+<ul class="companies">
+  <% @companies.each do |company| %>
+    <li>
+      Company: <%= company.name %><br/>
+      <%= link_to company.name, company %>
+    </li>
+  <% end %>
+</ul>
```
Comment (ryan-mok) [inline app/views/companies/index.html.erb]: Newline character at end of file

**Pull request merged**